### PR TITLE
fix: Re-fetch detached current user OrgUnits

### PR DESCRIPTION
**Pull request by netroms**
Fix issue with detached org units in [DHIS2-13287](https://jira.dhis2.org/browse/DHIS2-13287)

### CODE REVIEW

**Review by jbee — APPROVED** [review]

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/association/jdbc/JdbcOrgUnitAssociationStoreConfiguration.java [hunk 1]
```diff
@@ -46,28 +47,29 @@
 
     @Bean( "jdbcProgramOrgUnitAssociationsStore" )
     public JdbcOrgUnitAssociationsStore jdbcProgramOrgUnitAssociationStore( CurrentUserService currentUserService,
-        JdbcTemplate jdbcTemplate )
+        JdbcTemplate jdbcTemplate, OrganisationUnitService unitService )
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/association/jdbc/JdbcOrgUnitAssociationStoreConfiguration.java:50]: Could we name this `organisationUnitService`?

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/association/jdbc/JdbcOrgUnitAssociationsStore.java [hunk 2]
```diff
@@ -119,7 +122,14 @@ public SetValuedMap<String, String> getOrganisationUnitsAssociations( Set<String
 
     private Set<String> getUserOrgUnitPaths()
     {
-        Set<String> allUserOrgUnitPaths = currentUserService.getCurrentUserOrganisationUnits().stream()
+        Set<OrganisationUnit> currentUserOrganisationUnits = currentUserService.getCurrentUserOrganisationUnits();
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/association/jdbc/JdbcOrgUnitAssociationsStore.java:125]: Just curious, why is this necessary? Are we getting detached or cached org units from the current user service?

Comment (netroms) [reply dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/association/jdbc/JdbcOrgUnitAssociationsStore.java:125]: Yes, right now the user is detached after re-trivial on login. This is possible the disable very easy, right now I think we should consider this an ongoing experiment.
